src/encoded/static/components/audit.js

```diff
@@ -363,8 +363,11 @@ export const auditDecor = AuditComponent => class extends React.Component {
     }
 
     auditIndicators(audits, id, options) {
-        const { session, search } = options || {};
-        const loggedIn = !!(session && session['auth.userid']);
+        const { session, search, sessionProperties } = options || {};
+        const reactContextUser = sessionProperties ? sessionProperties.user : null;
+        const isAdmin = reactContextUser && reactContextUser.admin;
+        const isSubmitter = reactContextUser && reactContextUser.lab && reactContextUser.lab.status === 'current' && reactContextUser.submits_for && reactContextUser.submits_for.length > 0;
+        const isAuthorized = isAdmin || isSubmitter;
```

Comment:
I feel like this isAdmin/isSubmitter/isAuthorized pattern will be needed in many places, and the tests (particularly isSubmitter) aren’t obvious. I’m thinking they could be three functions in globals.js, or one function in globals.js returning three booleans as an object, or maybe done once in app.js and supplied in context so people can just use things like `reactContext.isSubmitter` or something like that.

Reply:
Actually, is this isAdmin correct? the `admin` property is a property of sessionProperty, not user.

Reply:
I addressed this in the code-refactor